src/loader/bgw_launcher.c

```diff
@@ -911,9 +911,17 @@ database_is_template_check(void)
 						"syscache")));
 
 	pgdb = (Form_pg_database) GETSTRUCT(tuple);
+
+	if (!pgdb->datallowconn)
+		ereport(ERROR,
+				(errmsg("background worker \"%s\" connected to database that does not allow "
```

Comment:
I think we should rephrase it a bit because it will not connect but try to connect to a database that don't allow connections

Reply:
I reworded it as "trying to connect to ...".